components/resolved_ts/src/resolver.rs

```diff
@@ -29,6 +29,27 @@ pub struct Resolver {
     stopped: bool,
 }
 
+impl std::fmt::Debug for Resolver {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let far_lock = self.lock_ts_heap.iter().next();
+        let mut dt = f.debug_tuple("Resolver");
+        dt.field(&format_args!("region={}", self.region_id));
+
+        if let Some((ts, keys)) = far_lock {
+            dt.field(&format_args!(
+                "far_lock={:?}",
+                keys.iter()
+                    // We must use Display format here or the redact won't take effect.
+                    .map(|k| format!("{}", log_wrappers::Value::key(k)))
```

Comment:
Does it leak data when we enable log redact?

Reply:
The `Display` implement of `log_wrappers::Value` would redact the key.

<img width="665" alt="image" src="https://user-images.githubusercontent.com/36239017/166862966-383c52b5-0260-4fd3-bf73-a6d412aadead.png">